### PR TITLE
Feat/data access

### PR DESCRIPTION
**Pull request by Ignaciomendezp**
# Version 2.0.0 🎉

### Context
​
This is a [Soyio package](https://www.npmjs.com/package/@soyio/soyio-widget?activeTab=readme) for web applications used for registration and authentication of identities.

### What is being done

​Commit by commit:

1. New events that end the flow are added.
2. TypeScript types are updated.
3. The readme is updated.
4. The version is incremented.

#### Before you merge...

> [!IMPORTANT]
> - [x] **Version Update**: Confirm that the `package.json` has been updated to reflect a new version that is higher than the current version on the [main branch](https://github.com/Soyio-id/soy-io-widget), which should be the same version that is available on [npm](https://www.npmjs.com/package/@soyio/soyio-widget).This step is crucial as it allows for an automatic release process.

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -25,22 +17,14 @@ export type SignatureAttemptProps = {
...
-  isSandbox?: boolean,
-  developmentUrl?: string,
-}
-
-export type AuthAttemptConfig = {
-  flow: 'authenticate',
-  configProps: AuthAttemptProps,
+export type DataAccessRequestConfig = {
+  flow: 'data_access',
+  configProps: DataAccessRequestProps,
```
Comment (MatiasMontagna) [inline src/types/index.ts:27]: podríamos aprovechar de meter el breaking change de `flow` a `request`  aprovechando que ya los clientes van a tener que hacer cambios jaja

**Review by MatiasMontagna — COMMENTED** [review]
Aprovecharía de pitearme los `Flow` si ya estamos mandando el paso a la versión `2.0.0` jaja

File: src/types/index.ts (hunk 1 above)
Comment (Ignaciomendezp) [reply src/types/index.ts:27]: Seee, me daba lata nomás porque hay que soportar los 2 parámetros desde el `widget_controller` 😂
Voy a hacer los cambios

Comment (Ignaciomendezp) [reply src/types/index.ts:27]: Mentira, no había que hacer eso

**Review by MatiasMontagna — COMMENTED** [review]
Creo que solo queda cambiar el `flowTemplateId` y estaríamos 🚀

**Review by MatiasMontagna — APPROVED** [review]
💎

**Pull request merged**